src/components/sidebar/sidebar.component.tsx

```diff
@@ -1,120 +1,123 @@
 
-  const joinWord = (value) => {
-    return value.replace(/\s/g, '');
-  };
-
-  const unspecifiedFields = useMemo(() => {
-    return (
-      Object.keys(values)
-        .filter((key) => key.endsWith('-unspecified'))
-        // find parent control
-        .map((key) => key.split('-unspecified')[0])
-        // factor-out those with values
-        .filter((key) => isEmpty(values[key]))
-        // return the unspecified control keys
-        .map((key) => `${key}-unspecified`)
-    );
-  }, [values]);
+  const unspecifiedFields = useMemo(
+    () =>
+      Object.keys(values).filter(
+        (key) => key.endsWith('-unspecified') && isEmpty(values[key.split('-unspecified')[0]]),
+      ),
+    [values],
+  );
 
   const handleClick = (selected) => {
-    const activeID = selected.replace(/\s/g, '');
-    setActiveLink(selected);
-    scrollIntoView(activeID);
+    const activeId = joinWord(selected);
+    scrollIntoView(activeId);
   };
 
   const markAllAsUnspecified = useCallback(
     (toggled) => {
-      if (toggled) {
-        unspecifiedFields.forEach((field) => {
-          values[field] = true;
-        });
-      } else {
-        unspecifiedFields.forEach((field) => {
-          values[field] = false;
-        });
-      }
-      setValues(values);
+      const updatedValues = { ...values };
+      unspecifiedFields.forEach((field) => {
+        updatedValues[field] = toggled;
+      });
+      setValues(updatedValues);
     },
-    [unspecifiedFields],
+    [unspecifiedFields, values, setValues],
   );
 
   return (
     <div className={styles.sidebar}>
-      {[...scrollablePages].map((page, index) => {
-        const isCurrentSelected = joinWord(page.label) === selectedPage;
+      {pages.map((page, index) => {
+        if (page.isHidden) return null;
+
+        const isCurrentlySelected = joinWord(page.label) === selectedPage;
         const hasError = pagesWithErrors.includes(page.label);
 
         return (
-          !page.isHidden && (
-            <div
-              aria-hidden="true"
-              className={classNames({
-                [styles.sidebarSectionErrorActive]: isCurrentSelected && hasError,
-                [styles.sidebarSectionActive]: isCurrentSelected && !hasError,
-                [styles.sidebarSectionError]: !isCurrentSelected && hasError,
-                [styles.sidebarSection]: !isCurrentSelected && !hasError,
-              })}
-              key={index}
-              onClick={() => handleClick(page.label)}>
-              <div className={styles.sidebarSectionLink}>{page.label}</div>
-            </div>
-          )
+          <div
+            aria-hidden="true"
+            className={classNames({
+              [styles.erroredSection]: isCurrentlySelected && hasError,
+              [styles.activeSection]: isCurrentlySelected && !hasError,
+              [styles.activeErroredSection]: !isCurrentlySelected && hasError,
+              [styles.section]: !isCurrentlySelected && !hasError,
```

Comment:
Removes the sidebar prefix from the classNames to avoid [context duplication](https://github.com/kettanaito/naming-cheatsheet#avoid-context-duplication).